### PR TITLE
Upgrade libp2p and stabilized the version

### PR DESCRIPTION
**Pull request by lescuer97**
This is an upgrade for libp2p, I did a previous PR but got mixed up with rebases

### CODE REVIEW

**Comment by lescuer97** [comment]
bors try

File: swap/src/cli/event_loop.rs [hunk 1]
```diff
@@ -167,12 +168,9 @@ impl EventLoop {
                             tracing::info!("Successfully closed connection to Alice");
                             return;
                         }
-                        SwarmEvent::UnreachableAddr { peer_id, address, attempts_remaining, error } if peer_id == self.alice_peer_id && attempts_remaining == 0 => {
-                            tracing::warn!(%address, "Failed to dial Alice: {}", error);
+                        SwarmEvent::OutgoingConnectionError { peer_id,  error } if matches!(peer_id, Some(alice_peer_id) if alice_peer_id == self.alice_peer_id) => {
+                            tracing::warn!( "Failed to dial Alice: {}", error);
 
-                            if let Some(duration) = self.swarm.behaviour_mut().redial.until_next_redial() {
```
Comment (binarybaron) [inline swap/src/cli/event_loop.rs:173]: Why was this removed?

**Review by binarybaron — COMMENTED** [review]
just left one small comment.

File: swap/src/cli/event_loop.rs (hunk 1 above)
Comment (lescuer97) [reply swap/src/cli/event_loop.rs:173]: this was a mistake thank you for catching it

Comment (binarybaron) [reply swap/src/cli/event_loop.rs:173]: Ah great

File: swap/src/cli/list_sellers.rs [hunk 2]
```diff
@@ -164,32 +165,32 @@ impl EventLoop {
                                 self.reachable_asb_address.insert(peer_id, address.clone());
                             }
                         }
-                        SwarmEvent::UnreachableAddr { peer_id, error, address, .. } => {
-                            if address == self.rendezvous_addr {
+                        SwarmEvent::OutgoingConnectionError { peer_id, error } => {
+                            if let Some(peer_id_rendevous) = peer_id {
```
Comment (binarybaron) [inline swap/src/cli/list_sellers.rs:169]: I think you are making some big changes to the logic here, not only refactoring. Not sure that was intended by you.

**Old:**
```rust
if address == self.rendezvous_addr {
    tracing::error!(
        "Failed to connect to rendezvous point at {}: {}",
        address,
        error
    );

    // if the rendezvous node is unreachable we just stop
    return Vec::new();
} else {
    tracing::debug!(
        "Failed to connect to peer at {}: {}",
        address,
        error
    );
    self.unreachable_asb_address.insert(peer_id, address.clone());

    match self.asb_quote_status.entry(peer_id) {
        Entry::Occupied(mut entry) => {
            entry.insert(QuoteStatus::Received(Status::Unreachable));
        },
        _ => {
            tracing::debug!(%peer_id, %error, "Connection error with unexpected peer")
        }
    }
}
```

New:
````rust
if let Some(peer_id_rendevous) = peer_id {
  tracing::error!(
      "Failed to connect to rendezvous point at {}: {}",
      self.rendezvous_addr,
      error
  );

  self.unreachable_asb_address.insert(peer_id_rendevous, self.rendezvous_addr.clone());

  match self.asb_quote_status.entry(peer_id_rendevous) { // <-- You moved this from the else block into the if block
      Entry::Occupied(mut entry) => {
          entry.insert(QuoteStatus::Received(Status::Unreachable));
      },
      _ => {
          tracing::debug!(%peer_id_rendevous, %error, "Connection error with unexpected peer")
      }
  }
} else {
  tracing::error!(
      "Failed to connect to rendezvous point at {}: {}",
      self.rendezvous_addr,
      error
  );
  // if the rendezvous node is unreachable we just stop
  return Vec::new(); // <-- You moved this from the if block to the else block
}
```

File: swap/src/cli/list_sellers.rs [hunk 3]
```diff
@@ -164,32 +165,32 @@ impl EventLoop {
...
                                         entry.insert(QuoteStatus::Received(Status::Unreachable));
                                     },
                                     _ => {
-                                        tracing::debug!(%peer_id, %error, "Connection error with unexpected peer")
+                                        tracing::debug!(%peer_id_rendevous, %error, "Connection error with unexpected peer")
                                     }
                                 }
+                            } else {
+                                tracing::error!(
+                                    "Failed to connect to rendezvous point at {}: {}",
```
Comment (binarybaron) [inline swap/src/cli/list_sellers.rs:188]: This should be "Failed to connect to peer at {}: {}". The peer is not the rendezvous point

**Review by binarybaron — CHANGES_REQUESTED** [review]
Left two more comments

**Comment by lescuer97** [comment]
bors try

**Comment by lescuer97** [comment]
bors try

**Comment by lescuer97** [comment]
there are conflicts becuase I merged with upstream after the [Identify protocol feature](https://github.com/comit-network/xmr-btc-swap/pull/996) was added, trying to solve them right now.

File: swap/src/cli/list_sellers.rs (hunk 3 above)
Comment (binarybaron) [reply swap/src/cli/list_sellers.rs:188]: Did I make a mistake here? Aren't you checking if its the rendezvous point in line 169 and if its not you will print this message?

**Review by binarybaron — CHANGES_REQUESTED** [review]
Did we make a swap on this branch already?

**Comment by lescuer97** [comment]
> Did we make a swap on this branch already?
I built and did one in my local machine against your ASB, but of course your ASB is not stabilized

**Comment by lescuer97** [comment]
@binarybaron because I rebased instead of doing a direct commit for a review I lost your comment (sorry). I'm now checking for the rendezvous using the peerId instead of the address(no longer available), you will see an additional else statement is just an in case for the CLI to not get stuck

File: swap/src/cli/list_sellers.rs [hunk 4]
```diff
@@ -164,30 +165,42 @@ impl EventLoop {
...
-                                    address,
+                                    &self.rendezvous_addr,
                                     error
                                 );
 
                                 // if the rendezvous node is unreachable we just stop
                                 return Vec::new();
+                            } else {
+                                tracing::error!(
+                                    "You connected to the wrong Peer: {} Error: {}",
```
Comment (binarybaron) [inline swap/src/cli/list_sellers.rs:182]: I was so confused because of the indentation :D

Do you know if these events are also called for connections with asbs?
If they are we didn't really fail to connect to a wrong peer but we failed to connect to an asb.

File: swap/src/cli/list_sellers.rs [hunk 5]
```diff
@@ -164,30 +165,42 @@ impl EventLoop {
...
+                                );
+                                     // if for some reason the peer is not the same it will return empty too.
+                                     // this is just for the CLI to not get stuck
+                                     return Vec::new();
+                            }
                             } else {
                                 tracing::debug!(
                                     "Failed to connect to peer at {}: {}",
-                                    address,
+                                    &self.rendezvous_addr,
```
Comment (binarybaron) [inline swap/src/cli/list_sellers.rs:193]: If these events are also called for connections with asbs, then we don't know if the address is `self.rendezvous_addr` right?

File: swap/src/cli/list_sellers.rs [hunk 6]
```diff
@@ -164,30 +165,42 @@ impl EventLoop {
...
                                     error
                                 );
-                                self.unreachable_asb_address.insert(peer_id, address.clone());
+                                self.unreachable_asb_address.insert(self.rendezvous_peer_id, self.rendezvous_addr.clone());
```
Comment (binarybaron) [inline swap/src/cli/list_sellers.rs:196]: Same thing. I think the peer might be an asb. We are adding a rendezvous point here to a list called `self.unreachable_asb_address ` which doesn't really make sense.

**Review by binarybaron — CHANGES_REQUESTED** [review]
Left some comments on the event handling issue. I'm still not really sure what's wrong there.  Maybe @delta1 can provide some insight when he reviews the PR

**Comment by binarybaron** [comment]
> @binarybaron because I rebased instead of doing a direct commit for a review I lost your comment (sorry). I'm now checking for the rendezvous using the peerId instead of the address(no longer available), you will see an additional else statement is just an in case for the CLI to not get stuck

Thank you! No worries

**Comment by delta1** [comment]
Looking good to me, please address the requested changes by @binarybaron and we can get this in!

**Comment by delta1** [comment]
bors try

File: swap/src/cli/list_sellers.rs (hunk 5 above)
Comment (delta1) [reply swap/src/cli/list_sellers.rs]: 100% right, I've updated the logic and added some more logic here: 7838442947c471ce6deb91f19e989bd636f6f076

File: swap/src/cli/list_sellers.rs (hunk 6 above)
Comment (delta1) [reply swap/src/cli/list_sellers.rs]: Agreed, fixed in same commit: 7838442947c471ce6deb91f19e989bd636f6f076

File: swap/src/cli/list_sellers.rs (hunk 2 above)
Comment (delta1) [reply swap/src/cli/list_sellers.rs]: Updated this to reflect the following cases: 
- failed to connect to the rendezvous, this is the case where we bail out
- failed to connect to a peer and have the peer_id in the event, record this peer_id in the unreachable list 
- failed to connect to a peer and peer_id is None, just log the error 

7838442947c471ce6deb91f19e989bd636f6f076

File: swap/src/cli/list_sellers.rs (hunk 3 above)
Comment (delta1) [reply swap/src/cli/list_sellers.rs]: Agreed with you and fixed in 7838442947c471ce6deb91f19e989bd636f6f076

File: swap/src/cli/list_sellers.rs (hunk 4 above)
Comment (delta1) [reply swap/src/cli/list_sellers.rs]: Yeah it's awful that rustfmt fails in the select! macro. Updated this indentation correctly here: dc68344b62db029809a4999c11a645238e7c6ad7 

In general, we want to keep as little logic as possible in the select macro and rather call a function with the event. I'll do that update sometime after this goes in so as not to complicate this PR too much.

**Comment by delta1** [comment]
I did test this successfully on testnet yesterday, I'm keen to get it in. Pushed some fixes regarding @binarybaron's comments which were spot on. 

Once the CI passes then I'm going to run a manual test again with multiple ASBs and a rendezvous to make sure that logic works right. 

In the meanwhile, libp2p is on version 0.46 but I'll do that upgrade down the line since it's a bit more complicated.

**Comment by lescuer97** [comment]
@delta1 did you test with both asb and cli?

File: swap/src/cli/list_sellers.rs (hunk 2 above)
Comment (binarybaron) [reply swap/src/cli/list_sellers.rs]: Looks good to me. I wasn't sure if I missed something here, but you're spot on!

File: swap/src/cli/list_sellers.rs (hunk 3 above)
Comment (binarybaron) [reply swap/src/cli/list_sellers.rs]: Thanks!

File: swap/src/cli/list_sellers.rs (hunk 6 above)
Comment (binarybaron) [reply swap/src/cli/list_sellers.rs]: Awesome, thanks!

**Comment by delta1** [comment]
> @delta1 did you test with both asb and cli?

Yes I ran a testnet swap locally with asb and swap built from this branch.

**Comment by binarybaron** [comment]
LGTM! I'm very keen to finally get this merged as well. I'll also do some manual testing but feel free to merge once you think you've tested enough.

**Comment by delta1** [comment]
> This should be backwards compatible, correct?

It should be - I'll try test that too

**Comment by lescuer97** [comment]
The RPC error that I reported on this [issue](https://github.com/comit-network/xmr-btc-swap/issues/1061) happened in a build of this pull request

**Comment by delta1** [comment]
> The RPC error that I reported on this [issue](https://github.com/comit-network/xmr-btc-swap/issues/1061) happened in a build of this pull request

I don't think it's related, the issue has occurred on 0.10.2 as well

**Comment by delta1** [comment]
I'll do some more testing this weekend but I think this is ready to go in

**Comment by delta1** [comment]
I'm not 100% that this is backwards compatible, and trying to test with asb 0.10.2 isn't great because of the lack of log output. 

So I plan to include this with the upgrade to Monero in #1081 instead

**Comment by delta1** [comment]
bors merge

**Review by delta1 — APPROVED** [review]
